vignettes/advs.Rmd

```diff
@@ -0,0 +1,491 @@
+  NA_character_, "BMIPCTL", "BMI-for-age percentile", 10,
+  NA_character_, "HDCSDS", "Head Circumference-for-age z-score", 11,
+  NA_character_, "HDCPCTL", "Head Circumference-for-age percentile", 12,
+  NA_character_, "WGTHSDS", "Weight-for-length/height Z-Score", 13,
+  NA_character_, "WGTHPCTL", "Weight-for-length/height Percentile", 14
+)
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+
+adsl_vars <- exprs(SEX, BRTHDTC, TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+advs <- vs %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  ) %>%
+  derive_vars_dt(
+    new_vars_prefix = "BRTH",
+    dtc = BRTHDTC
+  ) %>%
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = VSDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+advs <- advs %>%
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup %>% filter(!is.na(VSTESTCD)),
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  ) %>%
+  mutate(AVAL = VSSTRESN)
+
+advs <- advs %>%
+  mutate(
+    ATPTN = VSTPTNUM,
+    ATPT = VSTPT,
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "SCREENING 1" ~ "-1",
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+```
+
+## Derive Additional Variables for Anthropometric indicators {#derive_vars}
+
+In order to be able to compare against the reference files we need to
+know each child's age and length/height respectively at the time of each
+vital signs assessment. So these need to be first derived as variables
+in your ADVS.
+
+Remember to ensure that the unit of these matches that of the reference
+files metadata you are comparing against, i.e. "days" for age and "cm"
+for length/height in our examples here.
+
+Note that some of the standard reference files we use here have
+different units and intervals themselves. For example when you look at
+the "by Age" charts WHO provides this by day but CDC gives this every
+half month. So when converting all as days this still leaves the
+challenge of how to join when the patient's current age falls between an
+interval (e.g. between 24.5 and 25 months for CDC). Here in our package
+functions we took the simple approach to use the nearest age from the
+metadata, rather than apply some form of interpolation. An alternative
+method you could use would be to derive current age according to the
+exact same intervals as the metadata in order for an exact join, but in
+our examples we just kept this by days.
+
+### Derived Variables for Parameters by Age
+
+A variable for current analysis age could be achieved using the
+following code (you might first need to derive `BRTHDT` if not already
```

Comment:
Should we mention that BRTHDT must be not missing ? There are countries were the full birth date can't be collected in the CRF. it's to the programming team to impute it...

Reply:
sure, will explain